src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -137,12 +138,13 @@ private List<UUID> getUnauthorizedSources(
   public ResponseEntity<JobModel> createSnapshot(
       @Valid @RequestBody SnapshotRequestModel snapshotRequestModel) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
-    List<UUID> snapshotSourceDatasetIds =
-        snapshotService.getSourceDatasetIdsFromSnapshotRequest(snapshotRequestModel);
+    List<Dataset> datasets =
+        snapshotService.getSourceDatasetsFromSnapshotRequest(snapshotRequestModel);
+    List<UUID> snapshotSourceDatasetIds = snapshotService.getDatasetsIds(datasets);
     // TODO: auth should be put into flight
     List<UUID> unauthorized = getUnauthorizedSources(snapshotSourceDatasetIds, userReq);
```

Comment:
Can this also use a single source?
```suggestion
   Dataset dataset =
        snapshotService.getSourceDatasetsFromSnapshotRequest(snapshotRequestModel).get(0);
    UUID snapshotSourceDatasetId = dataset.getId();
    // TODO: auth should be put into flight
    List<UUID> unauthorized = getUnauthorizedSources(List.of(snapshotSourceDatasetId), userReq);
```